CMakeLists.txt

```diff
@@ -435,114 +484,92 @@ function(is_sdk_requested name result_var_name)
+      set(SWIFT_HOST_TRIPLE "aarch64-unknown-linux-gnu")
+    elseif("${SWIFT_HOST_VARIANT_ARCH}" MATCHES "(powerpc64|powerpc64le)")
+      set(SWIFT_HOST_TRIPLE "${SWIFT_HOST_VARIANT_ARCH}-unknown-linux-gnu")
+    elseif("${SWIFT_HOST_VARIANT_ARCH}" MATCHES "(armv6|armv7)")
+      set(SWIFT_HOST_TRIPLE "${SWIFT_HOST_VARIANT_ARCH}-unknown-linux-gnueabihf")
     else()
-      set(swift_can_crosscompile_stdlib TRUE)
+      message(FATAL_ERROR "Unable to calculate triple for linux host on ${SWIFT_HOST_VARIANT_ARCH}")
     endif()
+  endif()
 
-    is_sdk_requested(LINUX swift_build_linux)
-    if(swift_build_linux)
-      configure_sdk_unix(LINUX "Linux" "linux" "linux" "x86_64" "x86_64-unknown-linux-gnu")
-      set(SWIFT_PRIMARY_VARIANT_SDK_default "LINUX")
-      set(SWIFT_PRIMARY_VARIANT_ARCH_default "x86_64")
-    endif()
+  # Should we build the standard library for the host?
+  is_sdk_requested(LINUX swift_build_linux)
+  if(swift_build_linux)
+    configure_sdk_unix(LINUX "Linux" "linux" "${SWIFT_HOST_VARIANT}" "${SWIFT_HOST_VARIANT_ARCH}" "${SWIFT_HOST_TRIPLE}")
+  endif()
 
-    is_sdk_requested(ANDROID swift_build_android)
-    if(swift_build_android AND ${swift_can_crosscompile_stdlib})
-      configure_sdk_unix(ANDROID "Android" "android" "android" "armv7" "armv7-none-linux-androideabi")
-      # This must be set, as variables such as "${SWIFT_SDK_${sdk}_PATH}" are
-      # referenced in several other locations.
-      set(SWIFT_SDK_ANDROID_PATH "${SWIFT_ANDROID_SDK_PATH}")
+  # Compatible cross-compile SDKS for LINUX: ANDROID (arch always armv7)
+  is_sdk_requested(ANDROID swift_build_android)
+  if("${SWIFT_ANDROID_NDK_PATH}" STREQUAL "")
+    set(swift_can_crosscompile_stdlib_android FALSE)
+  else()
+    set(swift_can_crosscompile_stdlib_android TRUE)
+  endif()
 
-      set(SWIFT_PRIMARY_VARIANT_SDK_default "ANDROID")
-      set(SWIFT_PRIMARY_VARIANT_ARCH_default "armv7")
-    endif()
+  if(swift_build_android AND ${swift_can_crosscompile_stdlib_android})
+    configure_sdk_unix(ANDROID "Android" "android" "android" "armv7" "armv7-none-linux-androideabi")
+    # This must be set, as variables such as "${SWIFT_SDK_${sdk}_PATH}" are
+    # referenced in several other locations.
+    set(SWIFT_SDK_ANDROID_PATH "${SWIFT_ANDROID_SDK_PATH}")
 
-  # FIXME: This only matches ARMv6l (by far the most common variant).
-  elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "armv6l")
-    configure_sdk_unix(LINUX "Linux" "linux" "linux" "armv6" "armv6-unknown-linux-gnueabihf")
-    set(SWIFT_HOST_VARIANT_ARCH "armv6")
-    set(SWIFT_PRIMARY_VARIANT_ARCH_default "armv6")
-  # FIXME: This only matches ARMv7l (by far the most common variant).
-  elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "armv7l")
-    configure_sdk_unix(LINUX "Linux" "linux" "linux" "armv7" "armv7-unknown-linux-gnueabihf")
-    set(SWIFT_HOST_VARIANT_ARCH "armv7")
+    # FIXME: Is this correct? Should Android be the primary variant when it's a cross-compile target?
```

Comment:
Okay, I've updated it so that Android is only set as the primary variant if we aren't building for the host. We never actually get configured like that, but that's how we should behave if we were to be.